src/storage/media-search-store.js

```diff
@@ -66,21 +67,67 @@ export default class MediaSearchStore extends EventTarget {
     }
 
     searchParams.get("locale", navigator.languages[0]);
-    searchParams.set("source", URL_SOURCE_TO_TO_API_SOURCE[urlSource]);
+
+    let source;
+    if (urlSource === "avatars") {
+      // Avatar are special since we request avatars from a different source based on the facet.
+      const filter = new URLSearchParams(location.search);
+      source = filter.get("filter") === "my-avatars" ? "avatars" : "avatar_listings";
+    } else {
+      source = URL_SOURCE_TO_TO_API_SOURCE[urlSource];
+    }
+    searchParams.set("source", source);
+
+    if (source === "avatars") {
+      searchParams.set("user", window.APP.store.credentialsAccountId);
+    }
 
     const url = getReticulumFetchUrl(`/api/v1/media/search?${searchParams.toString()}`);
     if (this.lastSavedUrl === url) return;
 
-    const res = await fetch(url);
-    const result = await res.json();
+    const result = await this._fetchMedia(url, source);
+
     if (this.requestIndex != currentRequestIndex) return;
 
     this.result = result;
-    this.nextCursor = this.result.meta.next_cursor;
+    this.nextCursor = this.result.meta && this.result.meta.next_cursor;
     this.lastFetchedUrl = url;
     this.dispatchEvent(new CustomEvent("statechanged"));
   };
 
+  _fetchMedia = async (url, source) => {
+    const headers = { "Content-Type": "application/json" };
+    const credentialsToken = window.APP.store.state.credentials.token;
+    if (credentialsToken) headers.authorization = `bearer ${credentialsToken}`;
+
+    const res = await fetch(url, { method: "GET", headers });
+    const body = await res.text();
+
+    let result;
+    try {
+      result = JSON.parse(body);
+    } catch (e) {
+      result = body;
+    }
+
+    if (source === "avatar_listings" && result.entries.length === 0) {
```

Comment:
I think there are some other changes we are going to need to make to have featured avatars work, but it would be nice to be able to start setting them up in the admin interface, so we should avoid making the call at all unless you have a flag set or something.

Reply:
Oh. Okay, I'll put this behind a query parameter so that we can test it when we're preparing the actual set of featured avatars.